Support dd-trace `5.0.0`

Thank you for this useful library! We're currently using `nestjs-ddtrace@4.0.0` with `dd-trace` which now has a new release `5.0.0` available. [The breaking changes look minimal](https://github.com/DataDog/dd-trace-js/blob/master/MIGRATING.md). Please be so kind as to add that version as a supported peerDependency so we can stay on the latest `dd-trace`.

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: nestjs-ddtrace@4.0.0
npm ERR! Found: dd-trace@5.0.0
npm ERR! node_modules/dd-trace
npm ERR!   dd-trace@"5.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer dd-trace@"^4" from nestjs-ddtrace@4.0.0
npm ERR! node_modules/nestjs-ddtrace
npm ERR!   nestjs-ddtrace@"4.0.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: dd-trace@4.23.0
npm ERR! node_modules/dd-trace
npm ERR!   peer dd-trace@"^4" from nestjs-ddtrace@4.0.0
npm ERR!   node_modules/nestjs-ddtrace
npm ERR!     nestjs-ddtrace@"4.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/package.json
+++ b/package.json
@@ -34,7 +34,7 @@
   "peerDependencies": {
     "@nestjs/common": "^10.1.3",
     "@nestjs/core": "^10.1.3",
-    "dd-trace": "^4"
+    "dd-trace": "^5"
   },
   "devDependencies": {
     "@nestjs/cli": "^10.1.11",

